plugins/jobs/src/js/JobsOverview.tsx

```diff
@@ -10,98 +8,127 @@ import { graphqlObservable, componentFromStream } from "data-service";
-
-const JobsOverview = componentFromStream(
-  (props$: Observable<JobsOverviewProps>) => {
-    return combineLatest(filters$, props$, jobs$)
-      .map(([filters, props, jobs]) => {
-        function handleFilterChange(searchString: string) {
-          filters$.next({ filter: searchString, ...filters });
+const JobsOverview = componentFromStream<JobsOverviewProps>(
+  (props$: Subscribable<JobsOverviewProps>): Subscribable<React.ReactNode> => {
+    const jobsOverviewQuery: any = gql`
+      query {
+        jobs(
+          namespace: $namespace
+          filter: $filter
+          sortBy: $sortBy
+          sortDirection: $sortDirection
+        ) {
+          namespace
+          filteredCount
+          totalCount
+          nodes
+        }
+      }
+    `;
+
+    const isModalOpen$ = new BehaviorSubject<boolean>(false);
+
+    const filter$ = new BehaviorSubject<string>("");
+
+    (props$ as Observable<JobsOverviewProps>)
+      .map(props => props.setFilter)
+      .distinctUntilChanged()
+      .combineLatest(filter$)
+      .subscribe(([setFilter, filter]) => setFilter(filter));
+
+    const namespace$ = (props$ as Observable<JobsOverviewProps>)
+      .map(props => props.namespace)
+      .distinctUntilChanged();
+
+    const args$: Observable<JobsQueryArgs> = combineLatest(
+      namespace$,
+      filter$
+    ).map(([namespace, filter]) => ({
+      namespace,
+      filter,
+      sortBy: "ID" as SortOption,
+      sortDirection: "ASC" as SortDirection
+    }));
+
+    const jobs$: Observable<JobConnection> = args$
+      .sampleTime(250)
+      .switchMap((args: JobsQueryArgs) => {
+        return graphqlObservable(
+          jobsOverviewQuery,
+          JobModel,
+          args
+        ) as Observable<{ data: { jobs: JobConnection } }>;
+      })
+      .map(data => data.data.jobs);
+
+    return combineLatest(filter$, isModalOpen$, jobs$)
+      .map(([filter, isModalOpen, jobs]) => {
+        function handleFilterChange(filter: string) {
+          filter$.next(filter);
         }
         function resetFilter() {
-          filters$.next({});
+          handleFilterChange("");
+        }
+        function handleCloseModal() {
+          isModalOpen$.next(false);
+        }
+        function handleOpenModal() {
+          isModalOpen$.next(true);
         }
 
-        const root: JobTree = new JobTree(parseJobs(jobs.nodes));
-
-        // item is necessary for legacy jobtree implementation
-        let item: Job | JobTree;
-        if (props.params.id !== undefined) {
-          item = root.findItem(item => {
-            return (
-              item instanceof JobTree &&
-              item.id === decodeURIComponent(props.params.id || "")
-            );
-          });
-        } else {
-          item = root;
+        const modal = (
+          <JobFormModalContainer
+            open={isModalOpen}
+            onClose={handleCloseModal}
+          />
+        );
+
+        if (jobs.totalCount > 0) {
```

Comment:
I would like to have it the other way around and use the `if` for the exceptional case: that data is missing.

Reply:
I agree. Will change if I touch the file again - just copy pastes old code here 🤷‍♂️